dired.py

```diff
@@ -260,10 +261,11 @@ def traverse_tree(self, root, path, indent, tree, expanded):
 
         files = []
         index_files = []
-        for f in items:
+        for item in items:
+            f = item['file']
             new_path = join(path, f)
             dir_path = u'%s%s' % (new_path.rstrip(os.sep), os.sep)
-            check = isdir(new_path)
+            check = item['is_dir'] == True
```

Comment:
`item['is_dir']` is supposed to be boolean, so it could be `check = item['is_dir']`, or if you insist on explicitness then `check = item['is_dir'] is True`